### DISCUSSION TITLE
Changes not saving to njsPC

### DISCUSSION THREAD

**Discussion by flyboyusa** [tagyoureit/nodejs-poolController#560] [App and O/S Config Questions]
I have nodejs-poolController and nodejs-poolController-dashPanel installed and working pretty well. My problem is, when I remove a feature circuit or change a schedule, the changes don't stick. Everytime the dashboard re-loads the config. from the panel, the changes that I've made in the dashpanel are overwritten.

**Comment by rstrouse** [comment]
Do you have a control panel other than njsPC attached to the i7+3?  I would like to capture a replay when you delete a feature or schedule from that controller.   You don’t actually delete anything from IntelliTouch it is simply marked as invalid.

**Comment by flyboyusa** [comment]
I have an Intellitouch indoor wireless remote that quit communicating with the transceiver, that's what started me down this path of trying to get njsPC working.  I removed the power wire to the transceiver, so now it's off of the RS485 bus.  I rebooted the panel and njsPC and still have the problem.
Nothing seems to save.....schedules, features, etc....but I'm able to control items fine.  Also, on the RS485 tab, it hangs on "Getting Port Settings..." 
I'll try and screen capture

**Comment by flyboyusa** [comment]
Also, fyi, it's a i5+3S model, if that matters

**Comment by flyboyusa** [comment]
Here's a replay capture of the reload....
[Archive.zip](https://github.com/tagyoureit/nodejs-poolController/files/9227901/Archive.zip)

**Comment by flyboyusa** [comment]
Here's a capture of deleting the Feature called "ChlorineFea"
[Archive.zip](https://github.com/tagyoureit/nodejs-poolController/files/9227952/Archive.zip)

**Comment by flyboyusa** [comment]
I'm hoping @rstrouse or @tagyoureit has any suggestions for me to try or steps to troubleshoot my issue.
I love this piece of software, it's great, but I'm really hoping I can get this bug worked out!

**Comment by rstrouse** [comment]
Yes I will have some things for you to try shortly.

**Comment by rstrouse** [comment]
Ok pull njsPC and dashPanel.  Please hard refresh your browser then attempt to delete the feature.  I think I set all the bytes appropriately,  I do not have this panel so if you could try to delete the feature.

If this works I will tackle the schedule next.

**Reply by flyboyusa** [reply]
I pulled and then rebooted Pi.  I hard refreshed browser, then deleted the feature.  Then reloaded config using the "Reload Config" button under the hamburger menu.
The feature is permanently gone from the feature setup menu....but it still stays showing up on the front summary screen.

**Reply by flyboyusa** [reply]
Oh wait...I did a refresh of my browser and the summery screen is correct now.  So I think that's good.
The schedule keeps reverting back to strange times still

**Comment by rstrouse** [comment]
For the RS485 port tab what hardware are you running on?  

* Is this a Raspberry Pi, Mac, or PC?
* Which OS version are you running?

**Reply by flyboyusa** [reply]
Running Raspberry Pi

**Comment by rstrouse** [comment]
Please run each of the following commands and post the results here
node -v
ls /dev/tty*

**Reply by flyboyusa** [reply]
dave@raspberrypi:~ $ node -v
v14.20.0

dave@raspberrypi:~ $ ls /dev/tty*
/dev/tty    /dev/tty19  /dev/tty3   /dev/tty40  /dev/tty51  /dev/tty62
/dev/tty0   /dev/tty2   /dev/tty30  /dev/tty41  /dev/tty52  /dev/tty63
/dev/tty1   /dev/tty20  /dev/tty31  /dev/tty42  /dev/tty53  /dev/tty7
/dev/tty10  /dev/tty21  /dev/tty32  /dev/tty43  /dev/tty54  /dev/tty8
/dev/tty11  /dev/tty22  /dev/tty33  /dev/tty44  /dev/tty55  /dev/tty9
/dev/tty12  /dev/tty23  /dev/tty34  /dev/tty45  /dev/tty56  /dev/ttyAMA0
/dev/tty13  /dev/tty24  /dev/tty35  /dev/tty46  /dev/tty57  /dev/ttyprintk
/dev/tty14  /dev/tty25  /dev/tty36  /dev/tty47  /dev/tty58  /dev/ttyUSB0
/dev/tty15  /dev/tty26  /dev/tty37  /dev/tty48  /dev/tty59
/dev/tty16  /dev/tty27  /dev/tty38  /dev/tty49  /dev/tty6
/dev/tty17  /dev/tty28  /dev/tty39  /dev/tty5   /dev/tty60
/dev/tty18  /dev/tty29  /dev/tty4   /dev/tty50  /dev/tty61

**Reply by flyboyusa** [reply]
The RS485 connection tab within the dash panel now seems to be working fine....it will connect and show the the RS485 connection details like it supposed to.

**Reply by rstrouse** [reply]
Ok so now onto the schedules.  You mentioned strange times.  Are these your schedule times?
![image](https://user-images.githubusercontent.com/47839015/182648192-735c8963-7761-4599-9ce7-27c6cc5f9f5a.png)

**Reply by flyboyusa** [reply]
Yes...I want the Pool Light and Ldnscp Lts to both turn on a sunset...but it keeps reverting back to 2:09pm

**Reply by rstrouse** [reply]
You must supply a latitude and longitude for the Heliotrope to work.  This is under the general section.

**Reply by flyboyusa** [reply]
Ahh thanks, that seems to have helped the schedule.
Now when I try and open the Circuits tab, I get a constant spinning "Loading Options....." yellow box

**Reply by rstrouse** [reply]
Please pull dashPanel and hard refresh your browser.

**Reply by flyboyusa** [reply]
Now it reverts back to 9:32am when I save, re-load browser and re-load Config....

**Reply by flyboyusa** [reply]
I can set a hard coded time, like 9:00pm and it sticks.  But if I use Sunset, it reverts.

**Reply by rstrouse** [reply]
What does the sunset value look like in the poolState.json?

**Reply by flyboyusa** [reply]
[mode.txt](https://github.com/tagyoureit/nodejs-poolController/files/9255896/mode.txt)

**Comment by rstrouse** [comment]
What numbers did you use for latitude and longitude.  Do you by chance have them reversed?

**Reply by flyboyusa** [reply]
I found an error, I didn't have the negative (-) value in for the longitude, this seems to have fixed the issue, thanks!

The only other issue is now I try and hide the Circuit "AUX4" from the main page features list and it keeps popping back up?

**Reply by rstrouse** [reply]
Are you using the `Show as Feature` checkbox or are you relying on the circuit function in IntelliTouch?

**Reply by flyboyusa** [reply]
I'm un-checking the "Show as Feature" and it then disappears from the main screen as expected.  Then when I re-load config, it pops back onto the main screen.

**Reply by rstrouse** [reply]
I am going to have to think about this one.  Perhaps @tagyoureit has an idea.  The reason this occurs is because Touch controllers do not have this information stored on the panel.  When you reload the configuration from the OCP, it clears the current configuration and reloads it fresh meaning that any of the items that are extended by njsPC are cleared.  This includes the Sunrise/Sunset settings for your schedules.   

Perhaps this is no longer required given all of the data from the panel should be overwritten when the OCP responds to the requests.  For instance, IntelliCenter does not clear the configuration it simply overwrites the existing settings with anything fresh from the OCP.

**Reply by tagyoureit** [reply]
Actually, the inadvertent reset was only happening when manually clicking the Reload config button.  I added an override for *Touch to match Intellicenter so the reset doesn't wipe out the extended data.  Pull njsPC and it should stick if you manually load the config again.

**Reply by flyboyusa** [reply]
So far that seems to have done the trick, much appreciated!

**Comment by flyboyusa** [comment]
Question regarding the schedule start time called "Sunset" - I set this and it's stayed on 8:59pm ever since I set it.  This was the correct sunset time when I set it, but I haven't seen it change since.

Will this update on a predetermined frequency?

**Reply by tagyoureit** [reply]
Pull njsPC.  Sunrise and sunset aren't things that *Touch actually knows about.  I added a hook to check the start/end times every 30 mins (:00 and :30 on the hour).  

Also, just so we can keep things straight, please start a new issue/discussion after the previous issue is resolved.  It makes it much easier logistically to track and monitor the different issues.  Thx.

**Reply by flyboyusa** [reply]
Awesome, thanks for everything, much appreciated!  It is now set on "Sunset" as the start time....it doesn't show an actual start time, but I'm hoping it will continually update in the background.
I'll keep an eye on it over the next couple weeks and make sure it's tracking sunset correctly.
Thanks again.
